File: python/examples/get_document_download.py
import pprint
import os
import sys

sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from request import BaseRequest
from dotenv import dotenv_values

config = dotenv_values()

# Example of GET /documents/:document_id/download

class GetDocumentDownloadRequest(BaseRequest):
    def __init__(self):
        super().__init__()

    def get_document_download(self):
        path = f"/documents/{document_id}/download"
        return self.make_request("GET", path)

document_id = config['DOCUMENT_ID']

get_document_download_request = GetDocumentDownloadRequest()
document_download = get_document_download_request.get_document_download()

pprint.pprint(document_download)

File: javascript/examples/get_document_download.js
const baseRequest = require('../base_request.js').baseRequest
const path = require('path');
process.chdir(path.resolve(__dirname, '../../'));
require('dotenv').config();

// Example of GET /documents/:document_id/download

// The specified document is saved as a pdf in the 'documents' directory

const DOCUMENT_ID = process.env.DOCUMENT_ID;

token = baseRequest.request.token({ path: `/documents/${DOCUMENT_ID}/download` });

baseRequest.request.get({ path: `/documents/${DOCUMENT_ID}/download`, token: token });
